### PR TITLE
Improve Deployment Speed

### PR DESCRIPTION
**Pull request by oliverzy**
See https://github.com/shipitjs/shipit-deploy/issues/19

### CODE REVIEW

File: tasks/deploy/update.js [hunk 1]
```diff
@@ -32,8 +33,23 @@ module.exports = function (gruntOrShipit) {
...
       shipit.log('Create release path "%s"', shipit.releasePath);
-      return shipit.remote('mkdir -p ' + shipit.releasePath)
-      .then(function () {
+      function mkdir() {
+        return shipit.remote('mkdir -p ' + shipit.releasePath);
+      }
+
+      var returnedPromise;
+      if (shipit.config.copyReleaseBeforeUpdate) {
+        returnedPromise = shipit.remote(util.format('set -o pipefail && ' +
```
Comment (neoziro) [inline tasks/deploy/update.js]: We need to be sure that this line is correct, and please put it in a separated function to clarify code and to add a JSDoc comment on it to explain what it does.

Comment (neoziro) [inline tasks/deploy/update.js]:
```js
var promise;

if (shipit.config.copyReleaseBeforeUpdate)
  promise = copyPreviousRelease()
    .then(..)
    .catch(..);
else
  promise = mkdir();

return promise
  .then(...);
```

**Comment by timkelty** [comment]
I have an alternate PR here: https://github.com/shipitjs/shipit-deploy/pull/33

It uses some of the recently added methods for getting the previous release dir, so its a bit simpler.

**Comment by timkelty** [comment]
Should be fixed with https://github.com/shipitjs/shipit-deploy/pull/33

**Pull request closed without merging**